opencti-platform/opencti-front/src/components/ItemEntityType.tsx

```diff
@@ -28,28 +33,76 @@ const useStyles = makeStyles(() => ({
 
 interface ItemEntityTypeProps {
   entityType: string;
+  maxLength?: number;
   variant?: string;
+  showIcon?: boolean;
+  isRestricted?: boolean;
+  styles?: React.CSSProperties;
 }
 
 const ItemEntityType: FunctionComponent<ItemEntityTypeProps> = ({
   variant = 'inList',
+  maxLength,
   entityType,
+  showIcon = false,
+  isRestricted = false,
+  styles = {},
 }) => {
   const classes = useStyles();
   const { t_i18n } = useFormatter();
   const style = variant === 'inList' ? classes.chipInList : classes.chip;
 
-  const { isRelationship } = useSchema();
+  const { isRelationship: checkIsRelationship } = useSchema();
+  const isRelationship = checkIsRelationship(entityType);
+
+  const { palette: { mode } } = useTheme();
+  const theme = mode === 'dark'
+    ? ThemeDark()
+    : ThemeLight();
+  const getStyle = () => {
+    if (isRestricted) {
+      const restrictedColor = itemColor('Restricted');
+      return {
+        backgroundColor: theme.palette.background.default,
+        color: theme.palette.chip.main,
+        border: `1px solid ${restrictedColor}`,
+      };
+    }
+    return {
+      backgroundColor: theme.palette.background.default,
+      color: isRelationship ? theme.palette.primary.main : theme.palette.chip.main,
+      border: `1px solid ${isRelationship ? theme.palette.primary.main : itemColor(entityType)}`,
+    };
+  };
+  const getIcon = () => {
+    if (showIcon && !isRelationship) {
+      return (
+        <ItemIcon
+          variant="inline"
+          type={isRestricted ? 'restricted' : entityType}
+        />
+      );
+    }
+    return <></>;
+  };
+  const getLabel = () => {
+    if (isRestricted) return t_i18n('Restricted');
+    const label = t_i18n(isRelationship ? `relationship_${entityType}` : `entity_${entityType}`);
+    if (maxLength) return truncate(label, maxLength);
+    return label;
+  };
 
   return (
     <Chip
       classes={{ root: style }}
       style={{
-        backgroundColor: hexToRGB(itemColor(entityType), 0.08),
-        color: itemColor(entityType),
-        border: `1px solid ${itemColor(entityType)}`,
+        ...styles,
+        ...getStyle(),
```

Comment:
I would rather make the prop style capable of overwriting any default style (this is standard behavior), so:
```suggestion
        ...getStyle(),
        ...styles,
```